### PR TITLE
Added export by tripIds to export by ids

### PR DESCRIPTION
**Pull request by peetjvv**
- expanded export by ids to take in tripIds and only export the data
 pertinent to those ids
 - few small performance improvements for exporting data lookups
 - sorting exported data by name (where appropriate) since the ids are guids now

### CODE REVIEW

File: GTFS/GTFSWriter.cs [hunk 1]
```diff
@@ -162,23 +173,22 @@ public void Write(T feed, IEnumerable<string> idsToWrite, IEnumerable<IGTFSTarge
...
             calendarsToWrite = calendarsToWrite.OrderBy(x => x.ServiceId).ToList();
             fareAttributesToWrite = fareAttributesToWrite.OrderBy(x => x.FareId).ToList();
             fareRulesToWrite = fareRulesToWrite.OrderBy(x => x.RouteId).ToList();
             frequenciesToWrite = frequenciesToWrite.OrderBy(x => x.TripId).ToList();
-            routesToWrite = routesToWrite.OrderBy(x => x.Id).ToList();
-            stopsToWrite = stopsToWrite.OrderBy(x => x.Id).ToList();
-            stopTimesToWrite = stopTimesToWrite.OrderBy(x => x.TripId).ToList();
+            routesToWrite = routesToWrite.OrderBy(x => x.ToString()).ToList();
+            stopsToWrite = stopsToWrite.OrderBy(x => x.Name ?? x.Id).ToList();
+            stopTimesToWrite = stopTimesToWrite.OrderBy(x => x.StopSequence).OrderBy(x => x.TripId).ToList();
```
Comment (timminata) [inline GTFS/GTFSWriter.cs:189]: Surely you want to order by TripId first and then by stop sequence?

Comment (peetjvv) [reply GTFS/GTFSWriter.cs:189]: nope, this is correct
image

Comment (timminata) [reply GTFS/GTFSWriter.cs:189]: I think both would give the correct output, this just seems a bit strange to me and probably requires a lot more shuffling than doing it the other way round. The intermediate step has all stops with the same sequence next to each other which just seems weird. But it does work.

**Review by timminata — APPROVED** [review]

File: GTFS/GTFSWriter.cs (hunk 1 above)
Comment (timminata) [reply GTFS/GTFSWriter.cs:189]: Happy though to complete

Comment (peetjvv) [reply GTFS/GTFSWriter.cs:189]: I think sorting the other way round will give you stop times with sequences 111122223333 instead of 123123123123.

**Pull request merged**